libs/runtime/unit/platform_unit_test.cpp

```diff
@@ -610,6 +610,7 @@ TEST_CASE("epoch_test_stale_items", "[platform]")
 
 static auto provider_function = []() { return EBPF_SUCCESS; };
 
+#if !defined(CONFIG_BPF_JIT_DISABLED)
```

Comment:
nit: how did this work earlier for `native_only` tests?

Reply:
Trampoline code is present even in native only builds. The issue is that it's x64 only and not actually used in native only, hence #define it out.